src/spark/state.cpp

```diff
@@ -78,6 +79,45 @@ unsigned char GetNetworkType() {
         return ADDRESS_NETWORK_REGTEST;
 }
 
+bool IsPayToSparkAddress(const CScript& script)
+{
+    const spark::Params* params = spark::Params::get_default();
+    spark::Address addr(params);
+    return IsPayToSparkAddress(script, addr);
+}
+
+bool IsPayToSparkAddress(const CScript& script, spark::Address& addr)
+{   if (script.empty() || script[script.size()-1] != OP_SPARKMINT)
+        return false;
+    unsigned char network = spark::GetNetworkType();
+    unsigned char coinNetwork;
+
+    std::vector<unsigned char> vch(script.begin() + 2, script.end() - 1);
```

Comment:
This beg+2, end-1 extraction pattern is being used often. It has to be a named function.